OnboardPietroPieri/App.tsx

```diff
@@ -69,36 +70,73 @@ const styles = StyleSheet.create({
 });
 
 type SectionProps = PropsWithChildren<{
+  type?: string,
   title: string;
+  setLogin: React.Dispatch<React.SetStateAction<string>>;
 }>;
 
-function Section({children, title}: SectionProps): React.JSX.Element {
+
+
+function Section({title, setLogin, type}: SectionProps): React.JSX.Element {
+  const isPassword = type === 'password';
+
   return (
     <View>
       <Text style={styles.inputText}>{title}</Text>
       <TextInput
         style={styles.input}
-        title={title}
-        keyboardType="email-address"
-        autoCapitalize="none">
+        keyboardType={isPassword ? 'default' : 'email-address'}
+        autoCapitalize="none"
+        secureTextEntry={isPassword}
+        onChangeText={(text: React.SetStateAction<string>) => setLogin(text)}>
```

Comment:
Simplificando a tipagem na interface também permite simplificar aqui
```suggestion
        onChangeText={(text: string) => setLogin(text)}>
```
Ou, como os parâmetros da função são iguais aos da função onChangeText, também é possível simplificar um pouco mais
```suggestion
        onChangeText={setLogin}>
```

Reply:
aplicado